client/configurationManager.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+
+export interface ConfigurationOverrides {
+    pathToCa: string;
+}
+
+export class ConfigurationManager {
+    private static vscodeConfiguration: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration();
+
+    static getVscodeConfiguration(): vscode.WorkspaceConfiguration {
+        return ConfigurationManager.vscodeConfiguration;
+    }
+
+    static handleUpdateConfigurationEvent(): void {
+        vscode.window.showErrorMessage('handleUpdateConfigurationEvent');
```

Comment:
Just curious, what does this line actually do? It seems odds to show an error message when a user changes config, is it doing something else?

Reply:
This was just me verifying that when I changed a setting, it would immediately fire and the singleton would be updated.
If we decide to actually merge this in, this line will be removed.